ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -24,15 +30,169 @@ int TestListBuckets(const Aws::S3Crt::ClientConfiguration &config);
 int
 TestListBuckets(const Aws::S3Crt::ClientConfiguration &config)
 {
+    int ret = 0;
     S3Connection conn(config);
     std::vector<std::string> buckets;
-    if (!conn.ListBuckets(buckets))
-        return 1;
+    if (ret = conn.ListBuckets(buckets) != 0)
+        return ret;
 
     std::cout << "All buckets under my account:" << std::endl;
     for (const auto &bucket : buckets)
         std::cout << "  * " << bucket << std::endl;
-    return 0;
+
+    return ret;
+}
+
+/*
+ * TestListObjects --
+ *     Unit test for listing S3 objects under the first bucket in the associated AWS account. This
+ *     test assumes there are initially no objects with the prefix of "test_list_objects_" in the
+ *     bucket.
+ */
+int
+TestListObjects(const Aws::S3Crt::ClientConfiguration &config)
+{
+    S3Connection conn(config);
+
+    /* Temporary workaround to get a bucket to use. */
+    std::vector<std::string> buckets;
+    int ret;
+    if (ret = conn.ListBuckets(buckets) != 0)
+        return (0);
+    if (buckets.empty()) {
+        std::cout << "No buckets found in AWS account." << std::endl;
+        return (1);
+    }
+
+    const std::string firstBucket = buckets.at(0);
+    std::vector<std::string> objects;
+
+    /* Total objects to insert in the test. */
+    const int32_t totalObjects = 20;
+    /* Prefix for objects in this test. */
+    const std::string prefix = "test_list_objects_";
+    /* Parameter for getting single object. */
+    const bool listSingle = true;
```

Comment:
Do we always want to call `ListObjects` with `listSIngle` being `true`?

Reply:
The `listSingle` param defaults to `false` if it is not passed in, so I have just used this variable for when we want `listSingle` to be true, for better readability.